### ISSUE TITLE
Request

### ISSUE THREAD

**Issue by DutchUser** [mbax/VanishNoPacket#629]
Can you make this work with SimpleBroadcast?
(With some 'Tags' people can see vanished people)
SimpleBroadcast  developer reaction when i ask it him,

Hello,

unfortunately VanishNoPacket doesn't provide an interface which I could use to implement this.
I'm sorry - you could ask the plugin developer of VNP if he maybe adds it.

Kind regards,
KingDome24

**Comment by TheMolkaPL** [comment]
As @mbax said:

> 1) Mojang has not clarified their position on the legality of Bukkit and modding. While they publicly stated on twitter that Bukkit has a 'special relationship' they very clearly denied that same relationship in the letter they sent Wolvereness which he quotes in his DMCA takedown. Their COO made several statements about the project which did not match up with reality, which is scary. I do not feel safe continuing work on Bukkit plugins until Mojang makes it explicitly clear that it's safe.

> 2) There has not been and will never be a 1.8 release of Bukkit or CraftBukkit. The only major Bukkit fork legally updated to 1.8 is Glowstone, which still lacks a lot of features so isn't yet worth updating to even if the first item in this list were resolved. The other updated fork is Spigot, which I refuse to support. Here's a short list of examples why:

> They spent substantial time insulting me and my fellow Bukkit developers
The founder, md_5, regularly attacked me directly, out of some one-sided feud he created (example)
They spent a lot of time trying to convince the community they were doing the hard work of a Bukkit update, when they were just applying some patches on top of my hard work.
They convinced the community to pay them a huge pile of money to fight the DMCA takedown they received, only to never actually file the counterclaim they said they would do.
They are now ignoring the (apparently determined by the lawyers they paid to be valid, as they didn't issue a response) DMCA takedown, hiding their servers behind proxies.

#627 (comment)

**Comment by mbax** [comment]
VNP provides several ways in which another plugin can determine visibility.

* Metadata set on the player. Check the 'vanished' metadata. If it exists, and asBoolean is true, they are invis, otherwise visible.
* Various methods inside VNP are accessible, not advisable.
* Events are fired, not as advisable.

**Comment by DutchUser** [comment]
So you are not going update anymore?

Thx flor reaction to it

**Comment by Leescar** [comment]
I know this may not be the best place or way to approach this, however  I'm a tad selfish and foolish.
I opted to not open another ticket as this is not an issue and related to the principle of this ticket.

I wish to inquire that if the Bukkit issue becomes resolved or if Glowstone becomes something comprehensibly more workable, are you still considering continuing plugin development and projects? Rather has your interest been destroyed due to the cluster of how horrible things have tumbled into.

As in either developing and creating new plugins, possible collaborations and updating outdated ones?

I ask because I see you as an incredibly talented individual and as stated above I'm selfish and I really want to have hopes of your retention in the plugin scene, the amount of effort you have invested so far is admirable and I would look forward to seeing more works from you.

**Comment by mbax** [comment]
@DutchUser https://github.com/mbax/VanishNoPacket/issues/629#issuecomment-114640293

>So you are not going update anymore?
>
>Thx foor reaction to it

VNP won't see updates, but that doesn't mean it's broken! Latest release works fine, and the other plugin developer can set up his plugin to work fine with it! :)


@Leescar https://github.com/mbax/VanishNoPacket/issues/629#issuecomment-114649056
>I know this may not be the best place or way to approach this, however I'm a tad selfish and foolish.
>I opted to not open another ticket as this is not an issue and related to the principle of this ticket.
>
>I wish to inquire that if the Bukkit issue becomes resolved or if Glowstone becomes something comprehensibly more workable, are you still considering continuing plugin development and projects? Rather has your interest been destroyed due to the cluster of how horrible things have tumbled into.
>
>As in either developing and creating new plugins, possible collaborations and updating outdated ones?
>
>I ask because I see you as an incredibly talented individual and as stated above I'm selfish and I really want to have hopes of your retention in the plugin scene, the amount of effort you have invested so far is admirable and I would look forward to seeing more works from you.

Thank you for the very kind words about my work. :)

Bukkit may be gone (RIP), but that won't stop me from developing plugins or other fun stuff.

At this point in the post-Bukkit world, [Sponge](https://spongepowered.org/) appears to be the best hope the community has for a future. I've been contributing to the project in various ways, and have ported one plugin so far: [CraftIRC](https://forums.spongepowered.org/t/craftirc-an-irc-minecraft-relay-plugin/6083).

By port, I actually mean I decided the code wasn't good enough (and mostly not mine, in this case), and completely rewrote the plugin. I also didn't like the available IRC libraries, so I [wrote one of those too](https://github.com/KittehOrg/KittehIRCClientLib).

My free time is decreased from what I had while working on Bukkity things, but I'm still working on projects, slowly. As more of Sponge's API is functional I will be working on other plugins.

**Comment by Leescar** [comment]
Alright I will have to look into the running a server with Sponge, though as for now since the plugin list is small it will have to wait on an immediate switch for me. My players get bored and bitchy with vanilla after like a month, meh.
 I like how your working on an irc plugin, I consider it like one of the skeleton cores of a server for logging and shadow moderating. 
Though just have to be patient its not like an API can just become complete in such a short while, not with minecraft updating at such weird intervals.

**Comment by mbax** [comment]
>Alright I will have to look into the running a server with Sponge, though as for now since the plugin list is small it will have to wait on an immediate switch for me. My players get bored and bitchy with vanilla after like a month, meh.
>I like how your working on an irc plugin, I consider it like one of the skeleton cores of a server for logging and shadow moderating. 
>Though just have to be patient its not like an API can just become complete in such a short while, not with minecraft updating at such weird intervals.

Sponge is definitely not ready for everyone to switch over. With CraftIRC I only send chat data, no option of sending other information yet, because much of the API isn't ready. The API is built for much of it but the code won't actually fire or will fail.

While not at all properly documented, CraftIRC supports filtering and mangling of messages, which should be helpful for various server admin needs. Hopefully I'll get that documentation up and running 'soon'. :)

**Comment by DutchUser** [comment]
True. I use this plugin for weaks but never now about that you have problems with spigot ot Md :(
But you have planned Sponge Support when its come out?

**Comment by TheMolkaPL** [comment]
> But you have planned Sponge Support when its come out?

VanishNoPacket is a Bukkit plugin.

**Comment by DutchUser** [comment]
So? I can still ask it to him
And also its not even only bukkit it works also on an Sipgot Server

**Comment by TheMolkaPL** [comment]
VanishNoPacket is designed as a Bukkit plugin. It'll be to hard to support Sponge too. The best way is to write a second plugin only for Sponge.

**Comment by DutchUser** [comment]
...Then i change my question to.. Are you making an plugin that do te same as VanishNoPacket when Sponge come out?
* I ask it to mbax. Not to you. I not going react to you anymore (TheMolkaPL) and also VNP works without any issue on Spigot and is not only for Bukkit

**Comment by mbax** [comment]
@DutchUser https://github.com/mbax/VanishNoPacket/issues/629#issuecomment-115023667
>True. I use this plugin for weaks but never now about that you have problems with spigot or Md :( 
>But you have planned Sponge Support when its come out?

I have been discussing with various Sponge folks the concepts necessary for good Vanish support in Sponge. I plan on writing a VanishNoPacket for Sponge when the time is right. :)

@TheMolkaPL https://github.com/mbax/VanishNoPacket/issues/629#issuecomment-115028502
>VanishNoPacket is designed as a Bukkit plugin. It'll be to hard to code it for Sponge too. The best way is to write a second plugin from scratch only for Sponge.

I have written many plugins for Bukkit that can be transitioned over to another API very easily without rewriting 95% of the code. 

I will be rewriting VanishNoPacket for other reasons including not liking the old code and licensing concerns.

**Comment by Leescar** [comment]
Its sad to see the current bukkit fiasco is still going on.
 Had it not and had you more free time or the desire to continue work with bukkit plugins I would so selfishly attempt to <strike>conscript and bring you under slave labor</strike> discuss the idea of you to writing new plugins/ updating/cleaning code on others.

**Comment by mbax** [comment]
@Leescar https://github.com/mbax/VanishNoPacket/issues/629#issuecomment-115318486
>Its sad to see the current bukkit fiasco is still going on.
> Had it not and had you more free time or the desire to continue work with bukkit plugins I would so selfishly attempt to <strike>conscript and bring you under slave labor</strike> discuss the idea of you to writing new plugins/ updating/cleaning code on others. 

You monster! You won't take me alive!

As for Bukkit, it's not really a fiasco so much as we stopped development and Mojang said 'no' and then did nothing about it. Dead project is dead, our lives go on. :)

**Comment by Leescar** [comment]
But if your not alive how will you code? :x

**Comment by mbax** [comment]
>But if your not alive how will you code? :x
I'm alive, Bukkit is dead. :P

**Comment by EvilOlaf** [comment]
And we're glad to know....well, the first part at least ^^

**Comment by mbax** [comment]
I am closing this ticket as the original request has been resolved. Mostly to clear it out of my issue queue. ^_^

Discussion can, of course, continue here.
